.github/workflows/publish-js-client.yml

```diff
@@ -0,0 +1,146 @@
+name: Publish JS Client
+
+on:
+  workflow_dispatch:
+    inputs:
+      package_path:
+        description: Path to directory with package to release
+        required: true
+        default: 'clients/js'
+        type: choice
+        options:
+          - clients/js
+          - clients/js-legacy
+      level:
+        description: Version level
+        required: true
+        default: patch
+        type: choice
+        options:
+          - patch
+          - minor
+          - major
+          - prerelease
+          - prepatch
+          - preminor
+          - premajor
+      tag:
+        description: NPM Tag (and preid for pre-releases)
+        required: true
+        type: string
+        default: latest
+      create_release:
+        description: Create a GitHub release
+        required: true
+        type: boolean
+        default: true
+
+jobs:
+  test:
+    name: Test JS package
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Environment
+        uses: ./.github/actions/setup
+        with:
+          solana: true
+          cargo-cache-key: cargo-js-test-publish-${{ inputs.package_path }}
+          cargo-cache-fallback-key: cargo-js-test-publish
+
+      - name: Format
+        run: pnpm zx ./scripts/js/format.mjs "${{ inputs.package_path }}"
+
+      - name: Lint
+        run: pnpm zx ./scripts/js/lint.mjs "${{ inputs.package_path }}"
+
+      - name: Build Program
+        run: pnpm programs:build
+
+      - name: Test
+        run: pnpm zx ./scripts/js/test.mjs "${{ inputs.package_path }}"
+
+  publish:
+    name: Publish JS package
+    runs-on: ubuntu-latest
+    needs: test
+    permissions:
+      contents: write
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.ANZA_TEAM_PAT }}
+          fetch-depth: 0 # get the whole history for git-cliff
+
+      - name: Setup Environment
+        uses: ./.github/actions/setup
+
+      - name: Ensure SOLANA_NPM_TOKEN variable is set
+        env:
+          token: ${{ secrets.SOLANA_NPM_TOKEN }}
+        if: ${{ env.token == '' }}
```

Comment:
If I understand this correctly, the "solana-program" org does not need to have the SOLANA_NPM_TOKEN set. So this is requiring both secrets to be set, even though only one of them will be used.

Reply:
That's correct

Reply:
Sorry, so is this the desired behaviour (requiring both secrets)?

Reply:
Yeah that's how we do it with the other repos. I can remove the `SOLANA_PROGRAM_NPM_TOKEN` for now since this doesn't have a `@solana-program` package if you'd like. What do you think?